CMakeLists.txt

```diff
@@ -15,6 +15,35 @@ message(STATUS "Building the MPACT compiler at ${MPACT_SOURCE_DIR} (into ${MPACT
 
 set(MPACT_PYTHON_PACKAGES_DIR "${MPACT_BINARY_DIR}/python_packages")
 
+#-------------------------------------------------------------------------------
+# Configure out-of-tree vs in-tree build
+#-------------------------------------------------------------------------------
+
+if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
+  message(STATUS "Torch-MLIR out-of-tree build.")
+  message(FATAL_ERROR "TODO")
+else()
+  message(STATUS "Torch-MLIR in-tree build.")
+  # In-tree build with LLVM_EXTERNAL_PROJECTS=torch-mlir
+
+  option(MLIR_ENABLE_BINDINGS_PYTHON "Enables MLIR Python Bindings" OFF)
+
+  # TODO: Fix this upstream so that global include directories are not needed.
+  set(MLIR_MAIN_SRC_DIR ${LLVM_MAIN_SRC_DIR}/../mlir)
```

Comment:
can we make this MPACT specific VARs

the one you use are also used in the externals/ build set up

it probably does matter much, but it seems cleaner to have our own MPACT_ .... versions